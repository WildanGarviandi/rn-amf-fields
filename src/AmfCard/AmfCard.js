import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

class AmfCard extends Component {
  constructor(props) {
    super(props)
  }

  renderAction() {
    if (this.props.actionIcon) {
      return (
        <TouchableOpacity
          onPress={this.props.actionOnPress}
        >
          <Icon name={this.props.actionIcon} color={this.props.actionColor || 'black'} size={24}/>
        </TouchableOpacity>
      )
    }

    return null
  }

  renderTitle() {
    if (this.props.title) {
      return (
        <Text style={style.headerText}>{this.props.title}</Text>
      )
    }

    if (this.props.titleComponent) return this.props.titleComponent

    return (
      <View></View>
    )
  }

  renderHeader() {
    if (this.props.noHeader) return null

    return (
      <View style={style.headerContainer}>
        { this.renderTitle() }
        { this.props.actionComponent ? this.props.actionComponent : this.renderAction() }
      </View>
    )
  }

  render() {
    return (
      <TouchableOpacity
        style={[style.container, this.props.style]}
        onPress={this.props.onPress}
        disabled={this.props.disableTouch}
      >
        { this.renderHeader() }
        <View style={style.bodyContainer}>
          { this.props.body }
        </View>
      </TouchableOpacity>
    )
  }
}

const style = StyleSheet.create({
  bodyContainer: {
    padding: 16
  },
  container: {
    backgroundColor: 'white',
    margin: 8,
    borderRadius: 4,
    elevation: 4
  },
  headerContainer: {
    padding: 16,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 1,
    borderColor: '#E8E8E8'
  },
  headerText: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 16
  }
})

AmfCard.propTypes = {
  title: PropTypes.string,
  titleComponent: PropTypes.element,
  actionComponent: PropTypes.element,
  actionIcon: PropTypes.string,
  actionColor: PropTypes.string,
  actionOnPress: PropTypes.func,
  body: PropTypes.element.isRequired,
  noHeader: PropTypes.bool,
  onPress: PropTypes.func,
  style: View.propTypes.style,
  disableTouch: PropTypes.bool
}

export default AmfCard
