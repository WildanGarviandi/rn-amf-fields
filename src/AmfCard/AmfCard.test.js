import React from 'react'
import AmfCard from './AmfCard'
import renderer from 'react-test-renderer'

test('AmfCard without validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null
  }

  const component = renderer.create(<AmfCard {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


