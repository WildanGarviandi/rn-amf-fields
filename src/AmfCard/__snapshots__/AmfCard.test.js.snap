// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfCard without validation 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 4,
      "elevation": 4,
      "margin": 8,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#E8E8E8",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View />
  </View>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  />
</View>
`;
